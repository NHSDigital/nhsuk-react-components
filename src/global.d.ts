declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
