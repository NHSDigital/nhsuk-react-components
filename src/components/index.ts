export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
