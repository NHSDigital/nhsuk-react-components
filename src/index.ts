export * from './components';
export * from './patterns';
