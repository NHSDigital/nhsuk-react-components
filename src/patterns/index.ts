export * from './nav-a-z/index.js';
export * from './review-date/index.js';
