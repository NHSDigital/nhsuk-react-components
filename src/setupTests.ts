import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
