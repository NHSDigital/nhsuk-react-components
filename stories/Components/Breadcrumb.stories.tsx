import React from 'react';
import { Breadcrumb } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

/**
 * This component can be found in the `nhsuk-frontend` repository <a href="https://github.com/nhsuk/nhsuk-frontend/tree/main/packages/components/breadcrumb" target="_blank">here</a>.
 *
 *
 * ## Implementation Notes
 *
 * The `Breadcrumb` component contains two subcomponents: `Breadcrumb.Item` and `Breadcrumb.Back`. These are both anchor elements.
 *
 * The base `Breadcrumb` component has the default prop `aria-label` set to `"Breadcrumb"`. This is only a default prop and can be overridden.
 *
 * ## Usage
 *
 * ```jsx
 * import { Breadcrumb } from "nhsuk-react-components";
 *
 * const Link = () => {
 *     return (
 *         <Breadcrumb>
 *             <Breadcrumb.Item href="/level-one">Level one</Breadcrumb.Item>
 *             <Breadcrumb.Item href="/level-one/level-two">Level two</Breadcrumb.Item>
 *             <Breadcrumb.Item href="/level-one/level-two/level-three">Level three</Breadcrumb.Item>
 *             <Breadcrumb.Back href="/level-one/level-two/level-three">Back to Level three</Breadcrumb.Back>
 *         </Breadcrumb>
 *     );
 * }
 * ```
 */
const meta: Meta<typeof Breadcrumb> = {
  title: 'Components/Breadcrumb',
  component: Breadcrumb,
  render: (args) => (
    <Breadcrumb aria-label={args['aria-label']}>
      <Breadcrumb.Item href="/level/one">Level One</Breadcrumb.Item>
      <Breadcrumb.Item href="/level/two">Level Two</Breadcrumb.Item>
      <Breadcrumb.Item href="/level/three">Level Three</Breadcrumb.Item>
      <Breadcrumb.Back href="/level/three">Back to Level Three</Breadcrumb.Back>
    </Breadcrumb>
  ),
};
export default meta;
type Story = StoryObj<typeof Breadcrumb>;

Breadcrumb.Item.displayName = 'Breadcrumb.Item';
Breadcrumb.Back.displayName = 'Breadcrumb.Back';

export const Standard: Story = {};

export const OverrideAriaLabel: Story = {
  args: {
    'aria-label': 'custom-aria-label',
  },
};
