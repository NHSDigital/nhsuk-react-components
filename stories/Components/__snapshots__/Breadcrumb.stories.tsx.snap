// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Breadcrumb OverrideAriaLabel smoke-test 1`] = `
<nav class="nhsuk-breadcrumb"
     aria-label="custom-aria-label"
>
  <div class="nhsuk-width-container">
    <ol class="nhsuk-breadcrumb__list">
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/one"
        >
          Level One
        </a>
      </li>
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/two"
        >
          Level Two
        </a>
      </li>
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/three"
        >
          Level Three
        </a>
      </li>
    </ol>
    <p class="nhsuk-breadcrumb__back">
      <a class="nhsuk-breadcrumb__backlink"
         href="/level/three"
      >
        Back to Level Three
      </a>
    </p>
  </div>
</nav>
`;

exports[`Components/Breadcrumb Standard smoke-test 1`] = `
<nav class="nhsuk-breadcrumb"
     aria-label="Breadcrumb"
>
  <div class="nhsuk-width-container">
    <ol class="nhsuk-breadcrumb__list">
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/one"
        >
          Level One
        </a>
      </li>
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/two"
        >
          Level Two
        </a>
      </li>
      <li class="nhsuk-breadcrumb__item">
        <a class="nhsuk-breadcrumb__link"
           href="/level/three"
        >
          Level Three
        </a>
      </li>
    </ol>
    <p class="nhsuk-breadcrumb__back">
      <a class="nhsuk-breadcrumb__backlink"
         href="/level/three"
      >
        Back to Level Three
      </a>
    </p>
  </div>
</nav>
`;
